src/views/clusteroverview/SettingsTab/UserTab/components/ManageSSHKeySection/SSHAuthKeysList/hooks/useSSHAuthProjects.ts

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import {
+  modelToGroupVersionKind,
+  ProjectModel,
+  SecretModel,
+} from '@kubevirt-ui/kubevirt-api/console';
+import { getName } from '@kubevirt-utils/resources/shared';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import {
+  checkAccess,
+  K8sResourceCommon,
+  useK8sWatchResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+
+import { AuthKeyRow } from '../utils/types';
+
+type UseSSHAuthProjects = (authKeyRows: AuthKeyRow[]) => {
+  loaded: boolean;
+  selectableProjects: string[];
+};
+
+const useSSHAuthProjects: UseSSHAuthProjects = (authKeyRows) => {
+  const [projects, loaded] = useK8sWatchResource<K8sResourceCommon[]>({
+    groupVersionKind: modelToGroupVersionKind(ProjectModel),
+    isList: true,
+    namespaced: false,
+  });
+
+  const [selectableProjects, setSelectableProjects] = useState<string[]>([]);
+
+  const [loading, setLoading] = useState(true);
+
+  const selectedProjects = useMemo(
+    () =>
+      authKeyRows.map((row) => !isEmpty(row.projectName) && row.projectName).filter(Boolean) || [],
```

Comment:
why map and not only filter?

Reply:
because I need only the projectName and not the entire object :)

Reply:
in this case 1 reduce will make sure to combine both map and filter

Reply:
but its only n or 2n so it not really matter , if u dont want its also ok

Reply:
Changed to reduce method